src/System.Linq.Expressions/src/System/Linq/Expressions/BinaryExpression.cs

```diff
@@ -893,8 +893,7 @@ private static void ValidateUserDefinedConditionalLogicOperator(ExpressionType n
         private static void VerifyOpTrueFalse(ExpressionType nodeType, Type left, MethodInfo opTrue, string paramName)
         {
             ParameterInfo[] pmsOpTrue = opTrue.GetParametersCached();
-            if (pmsOpTrue.Length != 1)
-                throw Error.IncorrectNumberOfMethodCallArguments(opTrue, paramName);
+            Debug.Assert(pmsOpTrue.Length == 1);
```

Comment:
The method is retrieved by calling `TypeUtils.GetBooleanOperator(Type, string)`, found here: https://github.com/hughbe/corefx/blob/a82ade74aa16817e80e6df0bafaab47cbecd3204/src/System.Linq.Expressions/src/System/Dynamic/Utils/TypeUtils.cs#L703

This method calls `type.GetAnyStaticMethodValidated(name, new Type[] { type });`. This means that we will always only have one parameter!